File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0){
             return false;
              }
        int temp = x;
        long long reversed = 0;
        int remain = 0;
        while (temp != 0)
        {
            remain = temp % 10;
            reversed = reversed * 10 + remain;
            temp = temp / 10;
            
        }
        cout << reversed << endl;
        return reversed == x;
        }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False 
        y = str(x)
        n = len(y)
        for i in range(int(n+1/2)):
            if y[i] != y[n-1-i]:
                return False
        return True